tests/test_sos_model.py

```diff
@@ -292,6 +305,72 @@ def test_timestep_before(self):
         assert sos_model.timestep_before(2012) == 2011
         assert sos_model.timestep_before(2013) is None
 
+    def test_timestep_after(self):
+        sos_model = SosModel()
+        sos_model.timesteps = [2010, 2011, 2012]
+        assert sos_model.timestep_after(2010) == 2011
+        assert sos_model.timestep_after(2011) == 2012
+        assert sos_model.timestep_after(2012) is None
+        assert sos_model.timestep_after(2013) is None
+
+    def test_guess_outputs_zero(self, get_sos_model_only_scenario_dependencies):
+        """If no previous timestep has results, guess outputs as zero
+        """
+        sos_model = get_sos_model_only_scenario_dependencies
+        ws_model = sos_model.model_list['water_supply']
+
+        results = sos_model.guess_results(ws_model, 2010)
+        expected = {
+            "cost": [
+                SpaceTimeValue("oxford", 1, 0, "unknown")
+            ],
+            "water": [
+                SpaceTimeValue("oxford", 1, 0, "unknown")
+            ]
+        }
+        assert results == expected
+
+    def test_guess_outputs_last_year(self, get_sos_model_only_scenario_dependencies):
+        """If a previous timestep has results, guess outputs as identical
+        """
+        sos_model = get_sos_model_only_scenario_dependencies
+        ws_model = sos_model.model_list['water_supply']
+        expected = {
+            "cost": [
+                SpaceTimeValue("oxford", 1, 3.14, "unknown")
+            ],
+            "water": [
+                SpaceTimeValue("oxford", 1, 2.71, "unknown")
+            ]
+        }
+        sos_model._results[2010]['water_supply'] = [expected]
+        assert sos_model.timestep_before(2011) == 2010
+        results = sos_model.guess_results(ws_model, 2011)
+        assert results == expected
+
+    def test_converged_first_iteration(self, get_sos_model_only_scenario_dependencies):
+        """Should not report convergence after a single iteration
+        """
+        sos_model = get_sos_model_only_scenario_dependencies
+        ws_model = sos_model.model_list['water_supply']
+
+        results = sos_model.guess_results(ws_model, 2010)
+        sos_model._set_data(ws_model, 2010, results)
+
+        assert not sos_model.converged({'water_supply'}, 2010)
+
+    def test_converged_two_identical(self, get_sos_model_only_scenario_dependencies):
```

Comment:
Not exactly a flag - convergence can be assessed of a ModelSet for a given timestep. If there is just a single model in a ModelSet, we take a shortcut past any iteration or tests for convergence.